src/contrib/mlir/runtime/cpu/cpu_runtime.cpp

```diff
@@ -102,9 +107,19 @@ void MLIRCPURuntime::bindArguments(std::vector<void*>& externalTensors)
     // Assign external tensor pointers to invocation arguments.
     for (size_t i = 0, numArgs = m_invokeArgs.size(); i < numArgs; ++i)
     {
-        auto* memRefArg = *(reinterpret_cast<StaticMemRef**>(m_invokeArgs[i]));
-        memRefArg->allocatedPtr = (*m_externalTensors)[i];
-        memRefArg->alignedPtr = (*m_externalTensors)[i];
+        if (!clEnableCustomMemRefLowering)
+        {
+            // Default memref lowering lowers memrefs to StaticMemRef descriptors.
+            auto* memRefArg = *(reinterpret_cast<StaticMemRef**>(m_invokeArgs[i]));
+            memRefArg->allocatedPtr = (*m_externalTensors)[i];
+            memRefArg->alignedPtr = (*m_externalTensors)[i];
+        }
+        else
+        {
+            // Custom memref lowering lowers memrefs to plain pointers to tensors.
+            auto** memRefArg = reinterpret_cast<void**>(m_invokeArgs[i]);
```

Comment:
With custom memref lowering we still need to use struct to pass the layout information. It's needed for callbacks. https://github.com/NervanaSystems/ngraph/blob/171f5d4dcf5a57043f6ef7c21b62d3ad359385fc/src/contrib/mlir/runtime/cpu/cpu_runtime.cpp#L122